### PR TITLE
encodeQuery small improvement

### PR DESCRIPTION
**Pull request by juancarlospaco**
- `uri.encodeQuery` adds `usePrefix` boolean argument that prefixes a `"?"` at the start of the result string query.
- Not a huge change, but if you have to wrap this proc into another proc to do that then defeats practicality. 
- Small change is easy to merge. No api break because is a new argument.

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -68,6 +68,7 @@
 - `htmlgen.html` allows `lang` on the `<html>` tag and common valid attributes.
 - `macros.basename` and `basename=` got support for `PragmaExpr`,
   so that an expression like `MyEnum {.pure.}` is handled correctly.
+- `uri.encodeQuery` added `usePrefix` parameter prefixes a `?` at the start of the result string.
```
Comment (narimiran) [inline changelog.md]: > parameter prefixes

To me it sounds like a word is missing there. Maybe something like: _"...parameter which adds a prefix `?` ..."_

Comment (juancarlospaco) [reply changelog.md]: Done.

**Comment by juancarlospaco** [comment]
@narimiran Close or Merge as you see appropriate, no problem, is ready.

File: lib/pure/uri.nim [hunk 2]
```diff
@@ -147,6 +151,7 @@ proc encodeQuery*(query: openArray[(string, string)], usePlus = true,
     if not omitEq or val.len > 0:
       result.add('=')
       result.add(encodeUrl(val, usePlus))
+  if usePrefix and result.len > 0: result = "?" & result
```
Comment (Araq) [inline lib/pure/uri.nim:154]: So the implementation is just as inefficient as doing `` "?" & encodeQuery(x)``? Quite an achievement. :P

**Pull request closed without merging**